packages/nimble-components/src/rich-text/editor/tests/rich-text-editor.spec.ts

```diff
@@ -1052,6 +1052,395 @@ describe('RichTextEditor', () => {
+                        url: 'HttP://NimblE.NI.DEV'
+                    },
+                    {
+                        name: 'Absolute link(HttPS) in mixed case',
+                        input: '<a href="HttPS://NimblE.NI.DEV">HttPS://NimblE.NI.DEV</a>',
+                        url: 'HttPS://NimblE.NI.DEV'
+                    },
+                    {
+                        name: 'Hyper link(HttPS) in mixed case',
+                        input: '<a href="HttPS://NimblE.NI.DEV">Nimble</a>',
+                        url: 'HttPS://NimblE.NI.DEV'
+                    }
+                ] as const;
+
+                parameterizeNamedList(
+                    differentProtocolLinks,
+                    (spec, name, value) => {
+                        spec(
+                            `${name} renders as absolute link(href and text content should be same) in editor`,
+                            () => {
+                                pageObject.pasteHTMLToEditor(value.input);
+
+                                expect(pageObject.getEditorTagNames()).toEqual([
+                                    'P',
+                                    'A'
+                                ]);
+                                expect(
+                                    pageObject.getEditorLeafContents()
+                                ).toEqual([value.url]);
+                                expect(
+                                    pageObject.getEditorLastChildAttribute(
+                                        'href'
+                                    )
+                                ).toBe(value.url);
+                            }
+                        );
+                    }
+                );
+            });
+
+            describe('pasting various links between nodes should render as absolute links between nodes ', () => {
+                const differentProtocolLinks = [
+                    {
+                        name: 'Absolute link(https) within paragraph node',
+                        input: '<p>Anchor between <a href="https://nimble.ni.dev">https://nimble.ni.dev</a> text</p>',
+                        url: 'https://nimble.ni.dev',
+                        textContent: 'Anchor between https://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Hyperlink(http) within paragraph node',
+                        input: '<p>Anchor between <a href="http://nimble.ni.dev">Nimble</a> text</p>',
+                        url: 'http://nimble.ni.dev',
+                        textContent: 'Anchor between http://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Absolute link(https) as nimble-anchor within paragraph node',
+                        input: '<p>Anchor between <nimble-anchor href="https://nimble.ni.dev"><a href="https://nimble.ni.dev">https://nimble.ni.dev</a></nimble-anchor> text</p>',
```

Comment:
I'm confused by what is being tested here. Firstly, nesting `<a>` inside `<nimble-anchor>` is a strange and not particularly relevant use case, it seems (i.e. why would we ever encounter this?). Secondly, if you are going to test nested anchors, why aren't there any test cases that have `<nimble-anchor>` inside `<a>`? Thirdly, what if the `href` values of the `<a>` and `<nimble-anchor>` are different? Any why aren't there any test cases (anywhere in this file) of just `<nimble-anchor>` without a nested `<a>`? I'm not suggesting you add more test cases. I'm just trying to understand what about these cases that you're written are relevant. Would it be sufficient to just test `<nimble-anchor>` without the nested `<a>`?

Reply:
Basically, what am trying to test here is when copying a `nimble-anchor` from the viewer or any other source, in the clipboard, it stores as text/html as below.

<details><summary>HTML string when copied <code>nimble-anchor</code> from viewer</summary>
<p>

&lt;html&gt; &lt;body&gt; &lt;!--StartFragment--&gt;&lt;span style=&quot;color: rgb(0, 0, 0); font-family: &amp;quot;Source Sans Pro&amp;quot;, &amp;quot;Source Sans Pro Fallback&amp;quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(244, 244, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;&quot;&gt;&lt;span&gt; &lt;/span&gt;the&lt;span&gt; &lt;/span&gt;&lt;/span&gt;&lt;nimble-anchor href=&quot;https://nimble.ni.dev&quot; appearance=&quot;default&quot; style=&quot;display: inline; box-sizing: border-box; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-variant-position: ; font-weight: 400; font-stretch: ; font-size: 14px; line-height: ; font-family: &amp;quot;Source Sans Pro&amp;quot;, &amp;quot;Source Sans Pro Fallback&amp;quot;; font-optical-sizing: ; font-kerning: ; font-feature-settings: ; font-variation-settings: ; color: rgb(0, 0, 0); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(244, 244, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;&quot;&gt;&lt;a class=&quot;control&quot; part=&quot;control&quot; href=&quot;https://nimble.ni.dev/&quot; style=&quot;color: var(--ni-nimble-link-font-color); text-decoration: underline;&quot;&gt;&lt;span class=&quot;content&quot; part=&quot;content&quot;&gt;&lt;slot&gt;link&lt;/slot&gt;&lt;/span&gt;&lt;/a&gt;&lt;/nimble-anchor&gt;&lt;span style=&quot;color: rgb(0, 0, 0); font-family: &amp;quot;Source Sans Pro&amp;quot;, &amp;quot;Source Sans Pro Fallback&amp;quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(244, 244, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;&quot;&gt;&lt;span&gt; &lt;/span&gt;to n&lt;/span&gt;&lt;!--EndFragment--&gt; &lt;/body&gt; &lt;/html&gt;

</p>
</details> 

In the above HTML string, we can clearly see that the `<a>` tag is nested inside a `<nimble-anchor>`. So as to simulate the same pasting behavior of `nimble-anchor`, I just added a `<a>` within nimble-anchor which will fail if we remove the `transformPasted` logic from this code. 

Also, the case about having different `href` for nimble anchor and HTML anchor is technically not possible we did write that case.

Reply:
As discussed offline let's delete the nimble-anchor cases and instead file a tech debt item to track writing an integration test which instantiates a viewer and an editor, and copies from one to the other. That will be more robust if the format of the copied nimble anchor ever changes.

Reply:
Yes, I have removed the test cases related to `nimble-anchor` and created a tech debt here: https://github.com/ni/nimble/issues/1568